MediaBrowser.WebDashboard/dashboard-ui/bower_components/emby-webcomponents/browserdeviceprofile.js

```diff
@@ -1,384 +1,916 @@
-            h264Profiles = "high|main|baseline|constrained baseline";
-        maxH264Level >= 51 && browser.chrome && !browser.osx && (h264Profiles += "|high 10"), profile.CodecProfiles.push({
-            Type: "Video",
-            Codec: "h264",
-            Conditions: [{
-                Condition: "NotEquals",
-                Property: "IsAnamorphic",
-                Value: "true",
-                IsRequired: !1
-            }, {
-                Condition: "EqualsAny",
-                Property: "VideoProfile",
-                Value: h264Profiles
-            }, {
-                Condition: "LessThanEqual",
-                Property: "VideoLevel",
-                Value: maxH264Level.toString()
-            }]
-        }), browser.edgeUwp || browser.tizen || browser.orsay || browser.web0s, maxVideoWidth && profile.CodecProfiles[profile.CodecProfiles.length - 1].Conditions.push({
-            Condition: "LessThanEqual",
-            Property: "Width",
-            Value: maxVideoWidth.toString(),
-            IsRequired: !1
+
+        if (canPlayMkv && !browser.tizen && !browser.orsay && options.enableMkvProgressive !== false) {
+            profile.TranscodingProfiles.push({
+                Container: 'mkv',
+                Type: 'Video',
+                AudioCodec: videoAudioCodecs.join(','),
+                VideoCodec: mp4VideoCodecs.join(','),
+                Context: 'Streaming',
+                MaxAudioChannels: physicalAudioChannels.toString(),
+                CopyTimestamps: true
+            });
+        }
+
+        if (canPlayMkv) {
+            profile.TranscodingProfiles.push({
+                Container: 'mkv',
+                Type: 'Video',
+                AudioCodec: videoAudioCodecs.join(','),
+                VideoCodec: mp4VideoCodecs.join(','),
+                Context: 'Static',
+                MaxAudioChannels: physicalAudioChannels.toString(),
+                CopyTimestamps: true
+            });
+        }
+
+        if (canPlayHls() && options.enableHls !== false) {
+            profile.TranscodingProfiles.push({
+                Container: 'ts',
+                Type: 'Video',
+                AudioCodec: hlsVideoAudioCodecs.join(','),
+                VideoCodec: hlsVideoCodecs.join(','),
+                Context: 'Streaming',
+                Protocol: 'hls',
+                MaxAudioChannels: physicalAudioChannels.toString(),
+                MinSegments: browser.iOS || browser.osx ? '2' : '1',
+                BreakOnNonKeyFrames: hlsBreakOnNonKeyFrames
+            });
+        }
+
+        if (canPlayVp8) {
+            profile.TranscodingProfiles.push({
+                Container: 'webm',
+                Type: 'Video',
+                AudioCodec: 'vorbis',
+                VideoCodec: 'vpx',
+                Context: 'Streaming',
+                Protocol: 'http',
+                // If audio transcoding is needed, limit channels to number of physical audio channels
```

Comment:
It is strange that this channel forcing is not applied for all transcodings, only for some